clang_build/errors.py

```diff
@@ -3,6 +3,20 @@
 raised by clang-build if something goes wrong.
 '''
 
+class ApiError(RuntimeError):
```

Comment:
In general I am against creating new Error classes (as they are more cumbersome to work with (you have to import them)). I would only create a custom error class, if they provide additional functionality. This one seems to be a direct copy of CompileError. But the error_dict is not even used in api.py. Therefore, I think this one could just go.